examples/dreambooth/train_dreambooth_lora_sdxl.py

```diff
@@ -335,13 +389,41 @@ def parse_args(input_args=None):
             "Number of subprocesses to use for data loading. 0 means that the data will be loaded in the main process."
         ),
     )
+
+    parser.add_argument(
+        "--optimizer",
+        type=str,
+        default="prodigy",
+        help=(
+            'The optimizer type to use. Choose between ["adamW", "prodigy"]'
+        ),
+    )
+
     parser.add_argument(
-        "--use_8bit_adam", action="store_true", help="Whether or not to use 8-bit Adam from bitsandbytes."
+        "--use_8bit_adam", action="store_true", help="Whether or not to use 8-bit Adam from bitsandbytes. Ignored if "
+                                                     "optimizer is not set to AdamW"
     )
-    parser.add_argument("--adam_beta1", type=float, default=0.9, help="The beta1 parameter for the Adam optimizer.")
-    parser.add_argument("--adam_beta2", type=float, default=0.999, help="The beta2 parameter for the Adam optimizer.")
-    parser.add_argument("--adam_weight_decay", type=float, default=1e-2, help="Weight decay to use.")
-    parser.add_argument("--adam_epsilon", type=float, default=1e-08, help="Epsilon value for the Adam optimizer")
+
+    parser.add_argument("--adam_beta1", type=float, default=0.9,
+                        help="The beta1 parameter for the Adam and Prodigy optimizers.")
+    parser.add_argument("--adam_beta2", type=float, default=0.999,
```

Comment:
I have not tested this myself, but in [an issue of the Prodigy repo](https://github.com/konstmish/prodigy/issues/8#issuecomment-1804444174) some people reported using `adam_beta2=0.99` to work better when fine-tuning diffusion with LoRA.

Reply:
Thanks!